core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -271,7 +271,11 @@ oppia.factory('oppiaPlayerService', [
               classificationResult.ruleSpecIndex);
           }
 
-          var outcome = classificationResult.outcome;
+          // Use Object.assign to clone the object
+          // since classificationResult.outcome points
+          // at oldState.interaction.default_outcome
+          var outcome = Object.assign({}, classificationResult.outcome);
```

Comment:
Can we use angular.copy() instead? I think Object.assign() has compatibility issues with IE, so might as well avoid that if it's easy.

Also, great catch -- well done on debugging this!